ToolBox/GoogleDownloader/GoogleDataDownloader.cs

```diff
@@ -44,21 +39,122 @@ public class GoogleDataDownloader : IDataDownloader
         /// <returns>Enumerable of base data for this symbol</returns>
         public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime startUtc, DateTime endUtc)
         {
-            if (resolution != Resolution.Minute && resolution != Resolution.Hour)
-                throw new NotSupportedException("Resolution not available: " + resolution);
-
             if (symbol.ID.SecurityType != SecurityType.Equity)
                 throw new NotSupportedException("SecurityType not available: " + symbol.ID.SecurityType);
 
             if (endUtc < startUtc)
                 throw new ArgumentException("The end date must be greater or equal than the start date.");
 
+            switch (resolution)
+            {
+                case Resolution.Minute:
+                case Resolution.Hour:
+                    return GetMinuteOrHour(symbol, resolution, startUtc, endUtc);
+
+                case Resolution.Daily:
+                    return GetDaily(symbol, resolution, startUtc, endUtc);
+
+                default:
+                    throw new NotSupportedException("Resolution not available: " + resolution);
+            }
+        }
+
+        // q = SYMBOL
+        // startdate = Mon+D%2C+YYYY, e.g. Jan+1%2C+2005
+        // enddate = Mon+D%2C+YYYY, e.g. Sep+29%2C+2016
+        private const string UrlPrototypeDaily = @"http://www.google.com/finance/historical?q={0}&startdate={1}&enddate={2}&output=csv";
+        private enum ConvertMonths
+        {
+            Jan = 1,
+            Feb = 2,
+            Mar = 3,
+            Apr = 4,
+            May = 5,
+            Jun = 6,
+            Jul = 7,
+            Aug = 8,
+            Sep = 9,
+            Oct = 10,
+            Nov = 11,
+            Dec = 12
+        };
+
+        private IEnumerable<BaseData> GetDaily(Symbol symbol, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            var startdate = ((ConvertMonths)startUtc.Month).ToString()
+                + @"+" + startUtc.Day.ToString()
+                + @"%2C+" + startUtc.Year.ToString();
+            var enddate = ((ConvertMonths)endUtc.Month).ToString()
+                + @"+" + endUtc.Day.ToString()
+                + @"%2C+" + endUtc.Year.ToString();
+
+            // Create the Google formatted URL.
+            var url = string.Format(UrlPrototypeDaily, ConvertTicker(symbol), startdate, enddate);
+
+            // Download the data from Google.
+            string[] lines;
+            using (var client = new WebClient())
+            {
+                var data = client.DownloadString(url);
+                lines = data.Split('\n');
```

Comment:
I don't think this is an issue, but as a rule of thumb it is typically better to not directly split by lines, but instead use a stream reader and iterate by lines. In this case I suspect the 'file' is small enough that it doesn't matter, so this is just food for thought :)